src/main/java/com/fasterxml/jackson/databind/deser/std/StdKeyDeserializer.java

```diff
@@ -400,6 +421,8 @@ public Object _parse(String key, DeserializationContext ctxt) throws IOException
             }
             EnumResolver res = ctxt.isEnabled(DeserializationFeature.READ_ENUMS_USING_TO_STRING)
                     ? _getToStringResolver(ctxt) : _byNameResolver;
+            res = _byEnumNamingResolver == null ? res : _byEnumNamingResolver;
```

Comment:
this seems inefficient - if _byEnumNamingResolver != null then then there seems to be no reason to call this:

```
ctxt.isEnabled(DeserializationFeature.READ_ENUMS_USING_TO_STRING)
                    ? _getToStringResolver(ctxt) : _byNameResolver
```
How about this?
```
EnumResolver res _byEnumNamingResolver != null ? 
  _byEnumNamingResolver : 
  ctxt.isEnabled(DeserializationFeature.READ_ENUMS_USING_TO_STRING) ? 
    _getToStringResolver(ctxt) : _byNameResolver
```

Reply:
rewrite as you see fit - I just wanted to emphasise the logic to avoid confusion